binance_trade_bot/database.py

```diff
@@ -121,6 +121,19 @@ def get_pairs_from(self, from_coin: Union[Coin, str]):
             pairs: List[Pair] = session.query(Pair).filter(Pair.from_coin == from_coin)
             return pairs
 
+
+    def log_scout_stack(self, sh_stack):
+        session: Session
+        with self.db_session() as session:
+            merged_shs = []
+            for sh_temp in sh_stack:
+                pair = session.merge(sh_temp.pair)
+                sh = ScoutHistory(pair, sh_temp.target_ratio, sh_temp.current_coin_price, sh_temp.other_coin_price)
```

Comment:
Is there a reason we're creating a new object here? I'm sure there is because SQLAlchemy can sometimes be tricky but if there's another way we should do that.

```suggestion
                session.add(sh_temp)
```

Does that not work? What if you merged it first and then did that?

Reply:
`session.merge` takes an object that was created in another database session, refreshes it, and links it with the current session. We have to do this with pairs since they're created in another session.

You could do this:
```suggestion
                sh = session.merge(sh_temp)
```

and it might automatically merge the pair for you, I'm not sure. If that doesn't work, try:
```suggestion
                pair = session.merge(sh_temp.pair)
                sh_temp.pair = pair
```
Something like that? I can test it myself later if needs be

Reply:
Also don't know why it took more than a minute originally! That's a good speed increase

Reply:
Is this resolved? Does @j-waters modification work?

Reply:
I never ended up testing it, I can get round to it though

Reply:
running it for a couple of days. works fine.